src/cultionet/scripts/cultionet.py

```diff
@@ -740,128 +737,147 @@ def create_datasets(args):
+                image_vis=config['image_vis']
+            ).image_vis:
+                # Set the full path to the images
+                vi_path = ppaths.image_path / args.feature_pattern.format(
                     region=region,
-                    year=end_year,
-                    process_path=ppaths.get_process_path(args.destination),
-                    gain=args.gain,
-                    offset=args.offset,
-                    ref_res=ref_res,
-                    resampling=args.resampling,
-                    window_size=args.window_size,
-                    padding=args.padding,
-                    num_workers=args.num_workers,
-                    chunksize=args.chunksize
+                    image_vi=image_vi
                 )
-            else:
-                create_dataset(
-                    image_list=image_list,
-                    df_grids=df_grids,
-                    df_edges=df_edges,
-                    max_crop_class=args.max_crop_class,
-                    group_id=f'{region}_{end_year}',
-                    process_path=ppaths.get_process_path(args.destination),
-                    transforms=args.transforms,
-                    gain=args.gain,
-                    offset=args.offset,
-                    ref_res=ref_res,
-                    resampling=args.resampling,
-                    num_workers=args.num_workers,
-                    grid_size=args.grid_size,
-                    n_ts=args.n_ts,
-                    instance_seg=args.instance_seg,
-                    zero_padding=args.zero_padding,
-                    crop_column=args.crop_column,
-                    keep_crop_classes=args.keep_crop_classes,
-                    replace_dict=args.replace_dict
+
+                if not vi_path.is_dir():
+                    pbar.update(1)
+                    pbar.set_description('No directory')
+                    continue
+
+                # Get the centroid coordinates of the grid
+                lat = None
+                if args.destination != 'predict':
+                    lat = get_centroid_coords(df_grids.centroid, dst_crs='epsg:4326')[1]
+                # Get the start and end dates
+                start_date, end_date = get_start_end_dates(
+                    vi_path,
+                    start_year=end_year-1,
+                    start_date=args.start_date,
+                    end_date=args.end_date,
+                    date_format=args.date_format,
+                    lat=lat,
                 )
+                # Get the requested time slice
+                ts_list = model_preprocessing.get_time_series_list(
+                    vi_path,
+                    end_year-1,
+                    start_date,
+                    end_date,
+                    date_format=args.date_format
+                )
+                if len(ts_list) <= 1:
+                    pbar.update(1)
+                    pbar.set_description('TS too short')
```

Comment:
so hypothetically, if I only have `20210101.tif` and `20220101.tif` in my features, this function will continue?